File: guessNumber.java
package edu.ucsc;

/* The guess API is defined in the parent class GuessGame.
   @param num, your guess
   @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
      int guess(int num); */

public class guessNumber extends GuessGame {
    public int guessNumber(int n) {
        if (n == 0) {
            return 0;
        }
        long start = 1;
        long end = n;
        while (start + 1 < end) {
            long mid = start + (end - start) / 2;
            if (guess[mid] == 0) {
                return mid;
            } else if (guess[mid] == -1) {
                start = mid;
            }else {
                end = mid;
            }
        }
        return (int)start;
    }
}

File: Python/Binary_search/guessNumber.py
#! /Users/xiaotongli/anaconda3/bin/python
# -*- coding: utf-8 -*-
# @Time    : 9/28/18 10:57 PM
# @Author  : Xiaotong Li
# @School  : University of California, Santa Cruz
# @FileName: autocomplete_System.py
# @Software: PyCharm


# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        low, high = 1, n
        while low <= high:
            mid = (low + high) // 2
            res = guess(mid)
            if res == 0:
                return mid
            elif res == -1:
                right = mid - 1
            else:
                left = mid + 1
